compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -474,6 +478,72 @@ fn subst_and_check_impossible_predicates<'tcx>(
     result
 }
 
+/// Checks whether a trait's method is impossible to call on a given impl.
+///
+/// This only considers predicates that reference the impl's generics, and not
+/// those that reference the method's generics.
+fn is_impossible_method<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    (impl_def_id, trait_item_def_id): (DefId, DefId),
+) -> bool {
+    struct ReferencesOnlyParentGenerics<'tcx>(TyCtxt<'tcx>, &'tcx ty::Generics, DefId);
```

Comment:
i would prefer using named fields here, it wasn't immediately clear to me what this `DefId` should represent